File: AtCoder/ABC/C/274/C.py
n = int(input())
a = list(map(int, input().split()))
d = dict()
d[1] = 0
for i in range(1, n+1):
    d[2 * i] = d[a[i-1]] + 1
    d[2 * i + 1] = d[a[i-1]] + 1

for i in range(1, 2*n+2): print(d[i])

File: AtCoder/ABC/C/274/C.rb
n = gets.chomp.to_i
a = gets.chomp.split.map(&:to_i)
hash = Hash.new(0)
hash[1] = 0
a.each_with_index do |val, i|
    hash[2 * (i + 1)] = hash[val] + 1
    hash[2 * (i + 1) + 1] = hash[val] + 1
end
[*1..(2*n + 1)].each do |i| p hash[i]
end 